### ISSUE TITLE
needless_borrow false-positive when destructuring `&&mut T`

### ISSUE THREAD

**Issue by sinkuu** [Manishearth/rust-clippy#1432]
```rust
let mut s = String::new();
match &&mut s {
    &ref a => {} // WARNING: this pattern creates a reference to a reference (clippy)
}
```

```rust
let mut s = String::new();
match &&mut s {
    &a => {} // ERROR: cannot move out of borrowed content
}
```
